test/framework/debug.go

```diff
@@ -205,3 +233,114 @@ func CpRestarted(cluster Cluster) bool {
 		return false
 	}
 }
+
+// When we'll update our package to Go 1.23, below helper functions are can be replaced with
+// err = os.CopyFS(destDir, os.DirFS(srcDir))
+// ref#1. https://stackoverflow.com/a/56314145
+// ref#2. https://github.com/golang/go/issues/62484
+
+func CopyDirectory(scrDir, dest string) error {
+	entries, err := os.ReadDir(scrDir)
+	if err != nil {
+		return err
+	}
+	for _, entry := range entries {
+		sourcePath := filepath.Join(scrDir, entry.Name())
+		destPath := filepath.Join(dest, entry.Name())
+
+		fileInfo, err := os.Stat(sourcePath)
+		if err != nil {
+			return err
+		}
+
+		stat, ok := fileInfo.Sys().(*syscall.Stat_t)
```

Comment:
I guess we don't run tests on Darwin/Windows so this shouldn't matter?

Reply:
It works on darwin: https://github.com/golang/go/blob/master/src/syscall/ztypes_darwin_arm64.go#L65-L85

Reply:
I would love to get rid of these functions, but we can't currently smoothly upgrade to go 1.23 because of the version of golangci-linter. Tried to quickly update it, but it's more complex